### ISSUE TITLE
@versioned questions

### ISSUE THREAD

**Issue by austinamorusocfc** [aws-amplify/amplify-cli#273]
** Which Category is your question related to? **
GraphQL Transformer

** What AWS Services are you utilizing? **
Aws Appsynce GraphQL Transformer

** Provide additional details e.g. code snippets **
What happens with @versioned under the hood? If I'm using the apollo client and I have a fetch policy does the versioned effect any of the settings? 

type FetchPolicy =
  | 'cache-first'
  | 'cache-and-network'
  | 'network-only'
  | 'cache-only'
  | 'no-cache'
  | 'standby'

When is @versioned needed when is it not? 

Does @versioned cause the client to send up the version with the query then server to not return values if the version matches?

**Comment by undefobj** [comment]
@austinamorusocfc the @versioned sets up all the version checking for offline conflict resolution in the backend so that you can efficiently use the [AppSync SDK for Apollo](https://github.com/awslabs/aws-mobile-appsync-sdk-js#offline-helpers), especially with the [offline helpers](https://github.com/awslabs/aws-mobile-appsync-sdk-js/blob/master/OFFLINE_HELPERS.md). This allows you to do offline with automatic optimistic UI, etc., across multiple users and devices without having to wire up all the version checks in the backend yourself for CREATE/UPDATE/DELETE operations. These helpers only exist for the JavaScript SDK at the moment but you can still use this with the iOS and Android AppSync SDKs for Apollo too.

So in summary @versioned sets up the backend with the appropriate conditional checks for conflict resolution (you'll see things like `expectedVersion` in the GraphQL arguments) and the client helpers perform the client actions to send up the appropriate versions for different operations. You can of course also do these clients actions manually yourself but the helpers do it all in one line of code.

[Here is a tutorial of this in action](https://github.com/awslabs/aws-mobile-appsync-sdk-js/tree/master/tutorials/react-offline-realtime-todos) (we'll probably update this tutorial sometime to use the transformer as well).
